airflow/www/security.py

```diff
@@ -563,11 +561,11 @@ def add_homepage_access_to_custom_roles(self) -> None:
     def get_all_permissions(self) -> set[tuple[str, str]]:
         """Returns all permissions as a set of tuples with the action and resource names."""
         return set(
-            self.appbuilder.get_session.query(self.permission_model)
-            .join(self.permission_model.action)
-            .join(self.permission_model.resource)
-            .with_entities(self.action_model.name, self.resource_model.name)
-            .all()
+            self.appbuilder.get_session.execute(
+                select(self.action_model.name, self.resource_model.name)
+                .join(self.permission_model.action)
+                .join(self.permission_model.resource)
+            ).all()
```

Comment:
```suggestion
            )
```

This is also immediately turned into a set.

Reply:
fixed